src/components/AttachmentPicker/index.native.js

```diff
@@ -28,54 +35,72 @@ const documentPickerOptions = {
+        'ExpensifyCash does not have access to your camera. To enable access, tap Settings and turn on Camera.',
+        '',
+        [
+            {
+                text: 'Cancel',
+                style: 'cancel',
+            },
+            {
+                text: 'Settings',
+                onPress: () => Linking.openURL('app-settings:'),
+            },
+        ],
+        {cancelable: false},
+    );
+}
+
+/**
+ * A generic handling when we don't know the exact reason for an error
+ *
+ * @param {String} message
+ */
+function showGeneralAlert(message) {
+    Alert.alert(
+        'An error is preventing us to use handle the attachment',
+        message, // Todo: maybe we don't want to show this as it's probably not human friendly
+    );
+}
+
+/**
+ * Launch the DocumentPicker. Results are in the same format as ImagePicker
  *
- * @param {Object} callback
+ * @returns {Promise<DocumentPickerResponse>}
  */
-function showDocumentPicker(callback) {
-    RNDocumentPicker.pick(documentPickerOptions).then((results) => {
-        callback(results);
-    }).catch((error) => {
+function showDocumentPicker() {
+    return RNDocumentPicker.pick(documentPickerOptions).catch((error) => {
         if (!RNDocumentPicker.isCancel(error)) {
+            showGeneralAlert(error.message);
             throw error;
         }
     });
 }
 
 /**
- * Launch the AttachmentPicker. We display the ImagePicker first, as the document option is displayed as a
- * custom ImagePicker list item.
+ * Common image picker handling
  *
- * @param {Object} callback
+ * @param {function} imagePickerFunc - RNImagePicker.launchCamera or RNImagePicker.launchImageLibrary
+ * @returns {Promise<ImagePickerResponse>}
  */
-function show(callback) {
-    RNImagePicker.showImagePicker(imagePickerOptions, (response) => {
-        if (response.error) {
-            if (response.error === 'Camera permissions not granted') {
-                Alert.alert(
-                    // eslint-disable-next-line max-len
-                    'ExpensifyCash does not have access to your camera. To enable access, tap Settings and turn on Camera.',
-                    '',
-                    [
-                        {
-                            text: 'Cancel',
-                            style: 'cancel',
-                        },
-                        {
-                            text: 'Settings',
-                            onPress: () => Linking.openURL('app-settings:'),
-                        },
-                    ],
-                    {cancelable: false},
-                );
+function showImagePicker(imagePickerFunc) {
+    return new Promise((resolve, reject) => {
+        imagePickerFunc(imagePickerOptions, (response) => {
+            if (response.error) {
+                switch (response.error) {
+                    case 'Camera permissions not granted': showPermissionsAlert(); break;
+                    default: showGeneralAlert(response.error); break;
```

Comment:
Would you mind switching this to the default style that we use elsewhere? (with line breaks)

Reply:
I pushed a fix, but am leaving the comment unresolved so you can verify this fix properly address your comment